src/devices/inkbird_ith20r.c

```diff
@@ -102,7 +102,7 @@ static int inkbird_ith20r_callback(r_device *decoder, bitbuffer_t *bitbuffer)
     uint32_t subtype = (msg[3] << 24 | msg[2] << 16 | msg[1] << 8 | msg[0]);
     int sensor_num = msg[4];
     uint16_t word56 = (msg[6] << 8 | msg[5]);
-    int battery = msg[7];
+    float battery = msg[7] * 0.01f;
```

Comment:
I would suggest renaming to battery_ok as it is confusing to fill battery_ok with battery.    At least I at first thought you had a bug on review.

Also, it seems you believe that the 7-bit field has values from 0 to 100, but you are publishing eg only 0.1 and not 0.11, which is surprising.   I would add a comment that says what's in the field and why the precision is as it is.

Note that I like things explained to a far higher degree than others seem to ask for, but I stand by my opinion!

Reply:
The name is fixed -- thanks!   And the comment about comments, well I know I'm an outlier....

Reply:
I do want good documentation too. Hard to enforce, sometimes too bold comments about the nominal case can hide corner cases. E.g. here we have seen some values 0-100, I guess. But can't for certain claim to know there are no other special values. Maybe someday someone could map out all possible values with a variable voltage power supply?